test/kubernetes_unittest.cc

```diff
@@ -1951,4 +1966,52 @@ TEST_F(KubernetesTestFakeServerThreeWatchRetriesNodeLevelMetadata,
   EXPECT_TRUE(server->TerminateAllStreams(time::seconds(3)));
 }
 
+namespace {
+class MockSleeper : public Sleeper {
+ public:
+  MOCK_METHOD1(SleepFor, void(double seconds));
+};
+
+// Polls health_checker until it has at least one unhealthy component.
+// Returns false if newer timestamp not found after 3 seconds (polling
+// every 100 millis).
+bool WaitForUnhealthyComponents(const HealthChecker& health_checker) {
```

Comment:
I think having the timerange in function name will help reader reasoning - otherwise the reader needs to read through comments or code.

maybe `WaitForUnhealthComponentsAtLeast3Seconds` ? (If the function name is too long, come up with other names which makes sense...)

Reply:
Done.